service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -54,26 +53,26 @@ public static void writeCORSHeaders(
   }
 
   public static boolean isValidOrigin(String origin, CorsSupportProperties corsSupportProperties) {
+    if (corsSupportProperties.validHosts().contains("*")) {
+      return true;
+    }
+
     String stringToCheck;
     // We want to strip the protocol.
+    URL url;
     try {
-      URL url = new URL(origin);
-      stringToCheck = url.getHost();
+      url = new URL(origin);
     } catch (MalformedURLException e) {
-      stringToCheck = origin;
+      logger.info(e.toString());
```

Comment:
Would `logger.error` or `warn` be more appropriate here?

Reply:
Oh and can we get a wrapped bit of error text, like "Error parsing origin URL:" ? Will make it easier to search for if we ever need to debug from logs